### DISCUSSION TITLE
Welcome to fontbrowser Discussions!

### DISCUSSION THREAD

**Discussion by nianeyna** [nianeyna/fontbrowser#5] [Announcements]
Hello! Please feel free to post any questions, comments, feature requests, or bug reports you have that are related to fontbrowser in this forum (though you may also create an issue in the issues tab if you feel that is more appropriate to your concern). This program is currently in the early stages of development and is in a constant state of flux, so if you have an idea about it, please do speak up! I'm absolutely game to try pretty much anything at this point, so if you're thinking "wouldn't it be cool if-" then go ahead and drop that "if" in the chat and we'll see what we can do.

**Comment by Amyda1981** [comment]
Hello! Thank you for the update to the readme, I was able to download the .exe file for Windows.

So, I've opened the fontbrowser: for about 6-10 seconds I have the interface (font name search, character name search, etc.), but then it errors out with a statement "Oops! Sorry, and unexpected error has occurred. Invalid fond extension."

I used the View menu > reload, which does bring the interface back for a few seconds, but then the same error occurs. The program is not locked up, I can access menu options and close the program without a force quit.

I did reboot my computer to see if that settled things, but I still encountered the issue after reboot.

I'm not sure what other information might be helpful, please let me know if I can poke my computer in an informative manner for you.

Thank you for your work on this, I think it will be very neat to use!

**Reply by nianeyna** [reply]
You're actually the second person to have this problem and I *think* I figured out what it is - try the new version (v0.0.10) and see if that fixes it

**Reply by Amyda1981** [reply]
I finally got a chance to download again, things are working! Where did all these fonts come from, I didn't know I had all these?!?!

EDIT: oh, all the fonts i haven't actually installed are showing up too. Neat!

Thank you! <3

**Comment by e-hyde** [comment]
I opened fontbrowser and it would only let me scroll down through the first ~70 of my fonts, at which point the screen turned white and even scrolling up wouldn't show me the fonts that had been loaded before. I checked task manager and at this point it was using slightly less memory than firefox but it was rapidly going up, even though it was just sitting there with a blank screen (I closed the program when it was using twice as much as firefox but it wasn't showing signs of stopping). 

Is there a way to turn off infinite scroll and only load a page of fonts at a time? For reference, I'm not sure exactly how many fonts I have, but the first 70 hadn't even gotten me past the numbered fonts and to the ones that started with "A" yet.

Despite the complaint I also want to say I love the idea of the program, it has pretty much all the features that I would like to see in a font organizer. Thank you for making it!

**Reply by nianeyna** [reply]
Well that's no good. Do you have the latest version? For a brief period there was a version that had a non-optimized infinite scroll, so that could be it. 

If you definitely have the latest version, could you try opening the dev tools (either use ctrl+shift+i or View -> Toggle Developer Tools) before scrolling down. In the dev tools window there should be a tab called "Console" - open that, then try scrolling down again. After you've seen the behavior again (white screen etc) please right-click inside the "Console" display and select "Save As", then send me the resulting file. Thanks!

(Also I'm glad you like the features! Hopefully I can fix your problem so that you can actually use them, lol. Unfortunately it's probably not quite as simple as switching from infinite scroll to paging - under the hood the infinite scroll is actually doing basically the same thing as paging would performance-wise, so if you have this problem now, putting page numbers probably wouldn't fix it.)

**Reply by e-hyde** [reply]
[-1731297509955.log](https://github.com/user-attachments/files/17695017/-1731297509955.log)
[-1731298369003.log](https://github.com/user-attachments/files/17695018/-1731298369003.log)

I'm using 0.0.16, latest version unless I missed something

With the console open, I could see that it was a specific font that tripped it up. Since it was a dingbat font (2008 Noeuds from KPS Fonts) that I've never used and wouldn't miss, I deleted it to see if that solved the problem, and it did for a while, but now it has the same "failed to decode downloaded font" problem when I get to Bell MT (which I think came with windows). I attached the log for both times it happened. 

Thanks!

Also, played around with it a bit more, I could use the scroll bar to go down to the bottom and avoid Bell MT; I was able to see fonts further down in the alphabet that way until I got to a chinese font that caused the same problem.

**Reply by nianeyna** [reply]
This is really helpful, thanks so much! Question though, is the 2008 Noeuds a free font, and if so do you have a link to where I can download it? I tried googling but nothing came up.

**Reply by e-hyde** [reply]
Ah, it looks like the font maker's site is no longer online! https://web.archive.org/web/20240525052541/https://www.sch%C3%A4ffel.ch/de_gruppe4.html Here's the page I got it from, I think the downloads still work. I never used this one but I've used several other from the site without any issues.

**Reply by nianeyna** [reply]
Perfect, thank you! I believe I found the problem - try the new version I've just published and let me know if it works

**Reply by e-hyde** [reply]
It is not having the same problem! Bell MT (and a few other fonts, now that I can get past Bell MT to see them) still don't load, but they don't trip up the whole program anymore. 

However, I was scrolling through the full list, and when I got about halfway through, the rest of them stopped loading and it displayed a timeout error. When I restarted the program and started from the bottom, I could see the second half of my fonts just fine. I tried searching for some of the ones that didn't load, and it would just show the loading spiral--seems like any fonts that aren't loaded in the first five minutes won't show up at all unless I restart the program. Though, scrolling through my full list of fonts isn't really going to be a frequent use case once I get everything tagged.

Thank you so much for taking the time to look into this!

[-1731384238910.log](https://github.com/user-attachments/files/17711074/-1731384238910.log)

**Reply by nianeyna** [reply]
Oops. Yeah, in order to cut off the infinite loop that was causing your memory issue, I set a timeout of 5 minutes on font loading. I *thought* the clock would start ticking only once a font actually scrolled into view, but... clearly I was wrong. I'll keep working on it - thanks for your help!

**Reply by nianeyna** [reply]
new release up! let me know how it treats you

**Reply by e-hyde** [reply]
It still displayed a bunch of timeout error messages in the console, but the fonts load and everything seems to be working fine! Thanks!

**Comment by TheJellyCat** [comment]
Hi! I was curious if this is a project that you're still updating (I fully understand if you've moved on to other projects!). This seems to be about the only font organizer that will let me tag fonts without paying for premium access (which is still baffling) and it doesn't have a million features I'll never use, so it's very nearly perfect! I have a few fonts that won't load and would also love to know if tagging by font family is in the foreseeable future. I have... so many fonts... (It would also be neat to "roll up" the different styles under a font so I'm only looking at one per font if that makes sense? Then I can expand it if I wanna check out the italic or whatever.) 

Either way I very much love this for finding the perfect font amongst my hoard and want to say thanks for making it!

**Reply by nianeyna** [reply]
Well I haven't cracked it open in a while but that was mainly because I didn't think anyone was all that interested, I'm not *not* working on it. So by leaving this comment you have greatly increased your chances of getting your desired features implemented! I'll bump it up my priority list a bit. I'm currently taking a break from recreational coding though so it might be (wavy hand gesture) eh a couple months?
